5_data_munge/src/munge_data_files.R

```diff
@@ -0,0 +1,158 @@
+  dat <- readRDS(sc_retrieve(in_ind)) %>%
+    filter(!is.na(`ResultMeasure/MeasureUnitCode`)) %>%
+    filter(ActivityMediaName %in% 'Water') %>%
+    filter(!is.na(`ActivityDepthHeightMeasure/MeasureValue`)|
+                         !is.na(`ResultDepthHeightMeasure/MeasureValue`)|
+                         !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`)|
+                         !is.na(`ActivityBottomDepthHeightMeasure/MeasureValue`)) %>% 
+    filter(ActivityStartDate > as.Date('1800-01-01') & ActivityStartDate < as.Date('2020-01-01'))
+  
+  # create depth column
+  dat <- mutate(dat, sample_depth = case_when(
+    !is.na(`ActivityDepthHeightMeasure/MeasureValue`) ~ `ActivityDepthHeightMeasure/MeasureValue`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & !is.na(`ResultDepthHeightMeasure/MeasureValue`) ~ `ResultDepthHeightMeasure/MeasureValue`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityTopDepthHeightMeasure/MeasureValue`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityBottomDepthHeightMeasure/MeasureValue`
+  ))
+  
+  dat <- mutate(dat, sample_depth_unit_code = case_when(
+    !is.na(`ActivityDepthHeightMeasure/MeasureValue`) ~ `ActivityDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & !is.na(`ResultDepthHeightMeasure/MeasureValue`) ~ `ResultDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityTopDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityBottomDepthHeightMeasure/MeasureUnitCode`
+  ))
+  
+
+  # now reduce to daily measures
+  dat_reduced <- dat %>%
+    filter(ResultMeasureValue > min_value) %>%
+    filter(grepl('deg C|deg F', `ResultMeasure/MeasureUnitCode`, ignore.case = TRUE)) %>%
+    mutate(ResultMeasureValue = ifelse(grepl('deg F', `ResultMeasure/MeasureUnitCode`, ignore.case = TRUE),
+                                       f_to_c(ResultMeasureValue), ResultMeasureValue)) %>%
+    mutate(`ResultMeasure/MeasureUnitCode` = 'deg C') %>%
+    filter(ResultMeasureValue < max_value)
+    
+  dat_daily <- group_by(dat_reduced, MonitoringLocationIdentifier, ActivityStartDate, sample_depth) %>%
+    summarize(temperature_mean_daily = mean(ResultMeasureValue),
+              activity_start_times = paste(`ActivityStartTime/Time`, collapse = ', '),
+              n_day = n())
+  
+  dat_daily_meta <- filter(dat_reduced, !is.na(`ResultMeasure/MeasureUnitCode`)) %>%
```

Comment:
what case is handled by the `!is.na()` filter in this line? Note that
```
> grepl('deg C|deg F', NA)
[1] FALSE
```
so I'd expect that line 35 above would filter these out already

Reply:
Good catch - this is redundant.